src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/create/workflow/NewExperimentWorkflow.java

```diff
@@ -0,0 +1,107 @@
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@Getter
+@Singleton
+public class NewExperimentWorkflow implements ExperimentWorkflow {
+    private final ExperimentWorkflowNavigator.Workflow workflow;
+    private final ImportStatusService statusService;
+    private final Provider<ExperimentProcessor> experimentProcessorProvider;
+    private final Provider<ProcessorManager> processorManagerProvider;
+
+    @Inject
+    public NewExperimentWorkflow(ImportStatusService statusService,
+                                 Provider<ExperimentProcessor> experimentProcessorProvider,
+                                 Provider<ProcessorManager> processorManagerProvider){
+        this.statusService = statusService;
+        this.experimentProcessorProvider = experimentProcessorProvider;
+        this.processorManagerProvider = processorManagerProvider;
+        this.workflow = ExperimentWorkflowNavigator.Workflow.NEW_OBSERVATION;
+    }
+
+    @Override
+    public Optional<ImportWorkflowResult> process(ImportServiceContext context) {
+        // Metadata about this workflow processing the context
+        ImportWorkflow workflow = ImportWorkflow.builder()
+                .id(getWorkflow().getId())
+                .name(getWorkflow().getName())
+                .build();
+
+        // No-preview result
+        Optional<ImportWorkflowResult> result = Optional.of(ImportWorkflowResult.builder()
+                .workflow(workflow)  // attach metadata of this workflow to response
+                .importPreviewResponse(Optional.empty())
+                .caughtException(Optional.empty())
+                .build());
+
+        // Skip this workflow unless creating a new experiment
+        if (context != null && !this.workflow.isEqual(context.getWorkflow())) {
+            return Optional.empty();
+        }
+
+        // Skip processing if no context, but return no-preview result with metadata for this workflow
+        if (context == null) {
+            return result;
+        }
+
+        // Build and return the preview response
+        try {
+            ImportPreviewResponse successResponse;
+
+            // TODO: replace ProcessorManager#process with new-experiment workflow process from BI-2132
+            successResponse = processorManagerProvider.get().process(context.getBrAPIImports(),
+                    List.of(experimentProcessorProvider.get()),
+                    context.getData(),
+                    context.getProgram(),
+                    context.getUpload(),
+                    context.getUser(),
+                    context.isCommit());
+            result.ifPresent(importWorkflowResult -> importWorkflowResult.setImportPreviewResponse(Optional.of(successResponse)));
+        } catch (Exception e) {
+            result.ifPresent(importWorkflowResult -> importWorkflowResult.setCaughtException(Optional.of(e)));
+        }
+
```

Comment:
There isn't any need to change the signature of Workflow#process to include `throw Exception`. Once a call to the new-experiment pipeline replaces the old processor here, any lower level exceptions will be caught and handled fine.

Reply:
Accepted BI-2134 approach